openedx/core/djangoapps/xblock/utils.py

```diff
@@ -92,20 +92,8 @@ def _get_secure_token_for_xblock_handler(user_id, block_key_str, time_idx: int,
     Internal function to extract repeating hashing steps which we
     call mulitple times with different time_idx and hashing peppers.
     """
-    # This was supposed to be an interval boundry number that matches
-    # the current 2 day chunk(midnight UTC of every other day).
-    # Instead it's a number that will always be consistent but
-    # otherwise does not have a semantic meaning.
-    # Right now the math is missing a multiplication by TOKEN_PERIOD after
-    # the math.floor wich means the unit is days which is then added to
-    # 0 or -TOKEN_PERIOD seconds.
-    # Right now this is only valid for 0-2 days instead of the intended
-    # 2-4 days beacuse of the units issue above. Correcting the math should
-    # be possible but we are not going to do it in the middle of the other
-    # change we're making. We've made https://openedx.atlassian.net/browse/ARCHBOM-1677
-    # to come back and deal with it.
     TOKEN_PERIOD = 24 * 60 * 60 * 2  # These URLs are valid for 2-4 days
-    time_token = math.floor(time.time() / TOKEN_PERIOD)
+    time_token = math.floor(time.time() / TOKEN_PERIOD) * TOKEN_PERIOD
```

Comment:
This will break all current tokens because it will change the current value of time_token,  what about converting TOKEN_PERIOD into a days value instead so that all the current generated tokens remain valid but the previous window is also valid.

Reply:
Oof, completely forgot that.

Reply:
We can put in a test to pin the current token output and then confirm that it doesn't change for the time_idx=0 case. I like the idea of removing the mutliplication.